src/fetcher.ts

```diff
@@ -43,21 +42,21 @@ export interface ClientOptions {
 }
 
 export class BaseFetcher {
-  public readonly version: PriceEngineVersion = 'v9'
+  public readonly version: PriceEngineVersion
   private readonly apiKey: string
   private readonly baseURL: string
   private readonly timeout: number
   private readonly maxRetries: number
 
-  constructor(clientOptions: ClientOptions) {
-    this.baseURL = clientOptions?.baseURL || PRICE_ENGINE_BASE_URL
-    this.version = clientOptions?.version || this.version
-    this.timeout = clientOptions?.timeout || 5000
-    this.maxRetries = clientOptions?.maxRetries || 2
+  constructor(clientOptions: ClientOptions = {}) {
+    this.baseURL = clientOptions.baseURL || PRICE_ENGINE_BASE_URL
+    this.version = clientOptions.version || 'v9'
+    this.timeout = clientOptions.timeout || 5000
+    this.maxRetries = clientOptions.maxRetries || 2
 
     const apk = clientOptions?.apiKey || PRICE_ENGINE_API_KEY
     if (!apk || apk === undefined) {
```

Comment:
I think we should move this conditional block in the top of the constructor to avoid assign other variables if the api key not exist. What do you thin? @alonssrodd3